woocommerce/woocommerce
# Add mobile data to WCTracker

Adds a telemetry endpoint to record WCTracker data from the apps. We track each platform independently, so we can see which versions of the apps are used on a given site.

### All Submissions:






### Changes proposed in this Pull Request:

Adds a telemetry endpoint that we can use from the mobile apps to add mobile data usage to WCTracker.

### How to test the changes in this Pull Request:

1. Send a POST request to the `/wp-json/wc-telemetry/tracker` endpoint with body: `{ "platform": "ios", "version": "1.2" }`
    * Example (requires [the basic auth plugin](https://github.com/WP-API/Basic-Auth)): `curl -u user:password -X POST -H "Content-Type: application/json" -d '{ "platform": "ios", "version": "1.2" }' http://localhost/wp-json/wc-telemetry/tracker`
2. Via WP-CLI or similar, execute `WC_Tracker::get_tracking_data()`. Observe that `wc_mobile_usage` shows the correct platform and version.
    * Example:  `wp eval 'var_dump(WC_Tracker::get_tracking_data());'`
3. Send another POST request to the telemetry endpoint with body: `{ "platform": "ios", "version": "1.3" }`
4. Observe that `wc_mobile_usage` has recorded the updated version correctly.
5. Send another POST request to the telemetry endpoint with the original body: `{ "platform": "ios", "version": "1.2" }`
6. Observe that `wc_mobile_usage` has NOT updated with the older version number.

Note: If using `wp shell` to test, you may have to close the shell session and re-open if the option value is cached.

### Other information:


### Changelog entry

> Dev - Add mobile data to WCTracker

### FOR PR REVIEWER ONLY:
